File: android/src/main/kotlin/com/lucasjosino/on_audio_query/types/AudiosOnlyType.kt
package com.lucasjosino.on_audio_query.types

import android.os.Build
import android.provider.MediaStore

fun checkAudiosOnlyType(sortType: Int): String? {
    return when (sortType) {
        0 -> MediaStore.Audio.Media.IS_ALARM + " =1"
        1 -> MediaStore.Audio.Media.IS_MUSIC + " =1"
        2 -> MediaStore.Audio.Media.IS_NOTIFICATION + " =1"
        3 -> MediaStore.Audio.Media.IS_PODCAST + " =1"
        4 -> MediaStore.Audio.Media.IS_RINGTONE + " =1"
        5 -> if (Build.VERSION.SDK_INT >= 29) MediaStore.Audio.Media.IS_AUDIOBOOK + " =1" else null
        6 -> if (Build.VERSION.SDK_INT >= 29) MediaStore.Audio.Media.IS_PENDING + " =1" else null
        7 -> if (Build.VERSION.SDK_INT >= 30) MediaStore.Audio.Media.IS_FAVORITE + " =1" else null
        8 -> if (Build.VERSION.SDK_INT >= 30) MediaStore.Audio.Media.IS_DRM + " =1" else null
        9 -> if (Build.VERSION.SDK_INT >= 30) MediaStore.Audio.Media.IS_TRASHED + " =1" else null
        10 -> if (Build.VERSION.SDK_INT >= 30) MediaStore.Audio.Media.IS_DOWNLOAD + " =1" else null
        else -> null
    }
}

File: lib/details/types/audios_only_type.dart
part of on_audio_query;

/// Defines a specific type for query audios.
///
/// Note: The types below are specific for Android >= Q/29, so if you try call them below Android 10 will return [ALL_TYPE]
///
/// Types: [IS_AUDIOBOOK], [IS_PENDING], [IS_FAVORITE], [IS_DRM], [IS_TRASHED] and [IS_DOWNLOAD].
enum AudiosOnlyType {
  /// Query only audios defined with [IS_ALARM].
  IS_ALARM,

  /// Query only audios defined with [IS_MUSIC].
  IS_MUSIC,

  /// Query only audios defined with [IS_NOTIFICATION].
  IS_NOTIFICATION,

  /// Query only audios defined with [IS_PODCAST].
  IS_PODCAST,

  /// Query only audios defined with [IS_RINGTONE].
  IS_RINGTONE,

  /// Query only audios defined with [IS_AUDIOBOOK].
  ///
  /// Note: Works only in Android >= Q/29.
  IS_AUDIOBOOK,

  /// Query only audios defined with [IS_PENDING].
  ///
  /// Note: Works only in Android >= Q/29.
  IS_PENDING,

  /// Query only audios defined with [IS_FAVORITE].
  ///
  /// Note: Works only in Android >= Q/29.
  IS_FAVORITE,

  /// Query only audios defined with [IS_DRM].
  ///
  /// Note: Works only in Android >= Q/29.
  IS_DRM,

  /// Query only audios defined with [IS_TRASHED].
  ///
  /// Note: Works only in Android >= Q/29.
  IS_TRASHED,

  /// Query only audios defined with [IS_DOWNLOAD].
  ///
  /// Note: Works only in Android >= Q/29.
  IS_DOWNLOAD,

  /// Query only audios defined with [ALL_TYPE].
  ///
  /// It's a "DEFAULT".
  ALL_TYPE,
}
